lib/remote/apilistener-configsync.cpp

```diff
@@ -454,19 +502,17 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 
-	for (const Type::Ptr& type : Type::GetAllTypes()) {
-		auto *dtype = dynamic_cast<ConfigType *>(type.get());
-
-		if (!dtype)
-			continue;
-
-		for (const ConfigObject::Ptr& object : dtype->GetObjects()) {
-			/* don't sync objects for non-matching parent-child zones */
-			if (!azone->CanAccessObject(object))
-				continue;
-
-			/* send the config object to the connected client */
-			UpdateConfigObject(object, nullptr, aclient);
+	std::unordered_set<ConfigObject*> syncedObjects;
+	for (const Type::Ptr& type : Type::GetConfigTypesSortedByLoadDependencies()) {
```

Comment:
### Optional

The type of type is correct. However, I personally prefer

* either auto& x, if I want a reference of the same type (0692dcbc6f9c4941324cb46facf0b55cf4166a98)
* or T x, if I want to cast to T (#9731)

That avoids mistakes where you think you're just iterating by ref, but actually you're copying.